content/FileInclusion/FileInclusion.q.md

```diff
@@ -0,0 +1,1555 @@
+
+
+/// type=SS, answer=[3]
+
+On line 2 of `main.php`, replace `include` with `require`. Execute the program. What is its output?
+
+ - It prints `0`.
+
+ - It prints `2`.
+
+ - It prints `5`.
+
+ - It prints `10`.
+
+ - It produces an error.
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[main.php,DivideByZeroException.php,DivideByNegativeException.php]
+
+```php
+<?php
+require __DIR__ . "/DivideByZeroException.php";
+require_once __DIR__ . "/DivideByNegativeException.php";
+
+function divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        throw new DivideByZeroException("The value '0' as a divisor is not allowed.");
+    } elseif ($divisor < 0) {
+        throw new DivideByNegativeException();
+    } else {
+        return ($dividend / $divisor);
+    }
+}
+
+try {
+    echo divide(10, 2);
+} catch (DivideByZeroException $div) {
+    echo "Exception caught: " . $div->errorMessage();
+} catch (DivideByNegativeException $neg) {
+    echo "Exception caught: " . $neg->myErrorMessage() . " in " . $neg->myErrorFile();
+}
+?>
+```
+
+```php
+<?php
+class DivideByZeroException extends Exception
+{
+    public function errorMessage()
+    {
+        return $this->getMessage();
+    }
+}
+?>
+```
+
+```php
+<?php
+class DivideByNegativeException extends Exception 
+{
+    public function myErrorMessage()
+    {
+        return "A negative number as a divisor is encountered";
+    }
+
+    public function myErrorFile()
+    {
+        return $this->getFile();
+    }
+}
+?>
+```
+/// type=SS, answer=[3]
+
+On line 2 of `main.php`, enclose `__DIR__ . "/DivideByZeroException.php"` in parentheses `()`. Execute the program. What is its output?
+
+ - It prints `0`.
+
+ - It prints `2`.
+
+ - It prints `5`.
+
+ - It prints `10`.
+
+ - It produces an error.
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=SS, answer=[2]
+
+Which statement is true about file inclusion?
+
+ - It removes the specified file during the execution of a program.
+
+ - It enables the content of a certain file to be added to another PHP file.
+
+ - It adds and checks the specified file during the execution of a program.
+
+ - It is an object thrown by a program when there is an unexpected event or error that occurred.
+
+ - It is the process of detecting and dealing with unexpected events that occur in a program properly.
+
+
+/// type=MS, answer=[1,2,4,5]
+
+Which language constructs are used for file inclusion?
+
+ - `require`
+
+ - `include`
+
+ - `__DIR__`
+
+ - `include_once`
+
+ - `require_once`
+
+
+/// type=MS, answer=[2,5]
+
+Which statements correctly describe the `require` construct?
+
+ - It removes the specified file during the execution of a program.
+
+ - It adds and checks the specified file during the execution of a program.
+
+ - It adds and checks the content of the specified file to another file only once.
+
+ - It returns `E_WARNING` and continues the execution of a program if the specified file does not exist.
```

Comment:
This is not demonstrated in Part 1. Is it ok to have this here?

Reply:
Yes, since this is stated in our facts and this is on part 2. Whatever info that stated in our facts can be included in the q.